inference-engine/src/gna_plugin/gna_plugin.cpp

```diff
@@ -957,6 +957,10 @@ uint32_t GNAPlugin::QueueInference(const InferenceEngine::BlobMap &inputs, Infer
 }
 
 bool GNAPlugin::Wait(uint32_t request_idx) {
+    return WaitFor(request_idx, MAX_TIMEOUT);
```

Comment:
close to perfect - i think we can merge this way, optimal solution would be to expose wait wrapper over helper in a lambda:
see how std conditional wait designed:

``` c++
template< class Predicate >
void wait( std::unique_lock<std::mutex>& lock, Predicate pred );
```